drivers/place/template_mailer.cr

```diff
@@ -0,0 +1,224 @@
+require "placeos-driver"
+require "placeos-driver/interface/mailer"
+require "placeos-driver/interface/mailer_templates"
+
+class Place::TemplateMailer < PlaceOS::Driver
+  include PlaceOS::Driver::Interface::Mailer
+  include PlaceOS::Driver::Interface::MailerTemplates
+
+  descriptive_name "Template Mailer"
+  generic_name :Mailer
+  description %(uses metadata templates to send emails via the SMTP mailer)
+
+  default_settings({
+    cache_timeout: 300, # timeout for the template cache
+  })
+
+  accessor staff_api : StaffAPI_1
+
+  getter org_zone_ids : Array(String) { get_zone_ids?("org").not_nil! }
+  getter region_zone_ids : Array(String) { get_zone_ids?("region").not_nil! }
+  getter building_zone_ids : Array(String) { get_zone_ids?("building").not_nil! }
+  getter level_zone_ids : Array(String) { get_zone_ids?("level").not_nil! }
+
+  def mailer
+    system.implementing(Interface::Mailer)[1]
+  end
+
+  SEPERATOR = "."
+
+  @template_cache : TemplateCache = TemplateCache.new
+  @cache_timeout : Int64 = 300
+
+  def on_load
+    on_update
+  end
+
+  def on_update
+    @cache_timeout = setting?(Int64, :cache_timeout) || 300_i64
+
+    org_zone_ids.each do |zone_id|
+      update_template_fields(zone_id)
+    end
+  end
```

Comment:
@stakach 

Does it make sense to have it update the available fields on every org zone it can find, or should it be limited to where it's located `(zone_ids & system.zones).first`?

How about scheduling an update to avoid having to restart it every time a new driver with `template_fields` is added/removed

Reply:
I added a cron schedule for updating

Reply:
made it only update the org its in